pkg/util/fileutil/filesystem_test.go

```diff
@@ -266,6 +266,10 @@ func TestGetVolumeNameFromTargetPath(t *testing.T) {
 			targetPath: "/var/lib/kubelet/pods/7e7686a1-56c4-4c67-a6fd-4656ac484f0a/volumes/kubernetes.io~pv/pvvol/mount",
 			want:       "",
 		},
+		{
+			targetPath: "/var/lib/kubelet/pods/7e7686a1-56c4-4c67-a6fd-4656ac484f0a/volumes/kubernetesxio~csi/secrets-store-inline/mount",
+			want:       "",
+		},
```

Comment:
This gives `secrets-store-inline` without escaping the `.` which is fixed with this PR. Adding this test to validate the regex does a strict match.